examples/tox21/tox21_dataset_exploration.ipynb

```diff
@@ -0,0 +1,650 @@
+    "    NFPPreprocessor\n",
+    "\n",
+    "preprocessor = NFPPreprocessor()\n",
+    "train, val, test, train_smiles, val_smiles, test_smiles = datasets.get_tox21(preprocessor, labels=None, retain_smiles=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Dataset extraction depends on the `preprocessor`, and you may use other type of `preprocessor` as well.\n",
+    "\n",
+    "Below is another example of using `GGNNPreprocessor` for dataset extraction."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from chainer_chemistry.dataset.preprocessors.ggnn_preprocessor import \\\n",
+    "    GGNNPreprocessor\n",
+    "\n",
+    "#preprocessor = GGNNPreprocessor()\n",
+    "#train, val, test, train_smiles, val_smiles, test_smiles = datasets.get_tox21(preprocessor, labels=None, retain_smiles=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Check extracted dataset\n",
+    "\n",
+    "First, let's check number of data for train/validation/test dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "dataset information...\n",
+      "train <class 'chainer_chemistry.datasets.numpy_tuple_dataset.NumpyTupleDataset'> 11757\n",
+      "val <class 'chainer_chemistry.datasets.numpy_tuple_dataset.NumpyTupleDataset'> 295\n",
+      "test <class 'chainer_chemistry.datasets.numpy_tuple_dataset.NumpyTupleDataset'> 645\n",
+      "smiles information...\n",
+      "train_smiles <class 'list'> 11757\n"
+     ]
+    }
+   ],
+   "source": [
+    "print('dataset information...')\n",
+    "print('train', type(train), len(train))\n",
+    "print('val', type(val), len(val))\n",
+    "print('test', type(test), len(test))\n",
+    "\n",
+    "print('smiles information...')\n",
+    "print('train_smiles', type(train_smiles), len(train_smiles))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The dataset is a class of `NumpyTupleDataset`, where i-th dataset features can be accessed by `dataset[i]`.\n",
+    "\n",
+    "When `NFPPreprocessor` is used, each dataset consists of following features\n",
+    " 1. atom feature: representing atomic number of given molecule. \n",
+    " 2. adjacency matrix feature: representing adjacency matrix of given molecule.\n",
+    " 3. label feature: representing toxity (label) of given molecule.\n",
+    "    Here, 0 indicates negative (no toxity), 1 indicates positive (toxic) and -1 indicates data is not available, respectively."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's look an example of 7777-th train dataset"
```

Comment:
The result was different in my environment.

Reply:
let's discuss about it.